integration_tests/src/main/python/parquet_test.py

```diff
@@ -32,19 +32,27 @@ def read_parquet_sql(data_path):
     TimestampGen(start=datetime(1900, 1, 1, tzinfo=timezone.utc))],
     pytest.param([timestamp_gen], marks=pytest.mark.xfail(reason='https://github.com/NVIDIA/spark-rapids/issues/132'))]
 
+# test with original parquet file reader, the multi-file parallel reader for cloud, and coalesce file reader for
+# non-cloud
+original_parquet_file_reader_conf={'spark.rapids.sql.format.parquet.reader.type': 'PERFILE'}
+multithreaded_parquet_file_reader_conf={'spark.rapids.sql.format.parquet.reader.type': 'MULTITHREADED'}
+coalesce_parquet_file_reader_conf={'spark.rapids.sql.format.parquet.reader.type': 'COALESCING'}
+reader_opt_confs = [original_parquet_file_reader_conf, multithreaded_parquet_file_reader_conf,
+        coalesce_parquet_file_reader_conf]
+
 @pytest.mark.parametrize('parquet_gens', parquet_gens_list, ids=idfn)
 @pytest.mark.parametrize('read_func', [read_parquet_df, read_parquet_sql])
-@pytest.mark.parametrize('mt_opt', ["true", "false"])
+@pytest.mark.parametrize('reader_confs', reader_opt_confs)
 @pytest.mark.parametrize('v1_enabled_list', ["", "parquet"])
-def test_read_round_trip(spark_tmp_path, parquet_gens, read_func, mt_opt, v1_enabled_list):
+def test_read_round_trip(spark_tmp_path, parquet_gens, read_func, reader_confs, v1_enabled_list):
     gen_list = [('_c' + str(i), gen) for i, gen in enumerate(parquet_gens)]
     data_path = spark_tmp_path + '/PARQUET_DATA'
     with_cpu_session(
             lambda spark : gen_df(spark, gen_list).write.parquet(data_path),
             conf={'spark.sql.legacy.parquet.datetimeRebaseModeInWrite': 'CORRECTED'})
+    reader_confs.update({'spark.sql.sources.useV1SourceList': v1_enabled_list})
```

Comment:
nit: update modifies the existing dictionary. I believe that this means that we are literally modifying `original_parquet_file_reader_conf`, `multithreaded_parquet_file_reader_conf`, and `coalesce_parquet_file_reader_conf`.  In this case it does not matter because we set `spark.sql.sources.useV1SourceList` for every test every time, but to avoid configs/etc leaking between tests in the future I think it would be best if we first did a copy of the input config and then updated it.

Reply:
ok, if you are ok with it, I'll split the file and update the test in a separate PRs since this one is already large and hate to make you keep looking at it?

Reply:
scalastyle issue after accepting some suggestions and I didn't check, so I'll fix thee test here too

Reply:
Sorry, my bad, I didn't do the suggestion in an IDE but just inline in the comment. 😞